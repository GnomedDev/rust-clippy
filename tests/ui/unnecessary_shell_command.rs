#![warn(clippy::unnecessary_shell_command)]

use std::process::Command;

fn main() {
    let _ = Command::new("ls");
    //~^ error: unnecessarily shelling out for trivial operation
    let _ = Command::new("curl");
    //~^ error: unnecessarily shelling out for trivial operation
    let _ = Command::new("wget");
    //~^ error: unnecessarily shelling out for trivial operation
    let _ = Command::new("sed");
    //~^ error: unnecessarily shelling out for trivial operation
    let _ = Command::new("jq");
    //~^ error: unnecessarily shelling out for trivial operation

    let _ = Command::new("ffmpeg");
}
