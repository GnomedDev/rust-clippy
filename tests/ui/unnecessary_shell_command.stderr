error: unnecessarily shelling out for trivial operation
  --> tests/ui/unnecessary_shell_command.rs:6:26
   |
LL |     let _ = Command::new("ls");
   |                          ^^^^
   |
   = help: use `std::fs::read_dir` instead
   = note: `-D clippy::unnecessary-shell-command` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::unnecessary_shell_command)]`

error: unnecessarily shelling out for trivial operation
  --> tests/ui/unnecessary_shell_command.rs:8:26
   |
LL |     let _ = Command::new("curl");
   |                          ^^^^^^
   |
   = help: use `std::net`, `ureq`, or `reqwest` instead

error: unnecessarily shelling out for trivial operation
  --> tests/ui/unnecessary_shell_command.rs:10:26
   |
LL |     let _ = Command::new("wget");
   |                          ^^^^^^
   |
   = help: use `std::net`, `ureq`, or `reqwest` instead

error: unnecessarily shelling out for trivial operation
  --> tests/ui/unnecessary_shell_command.rs:12:26
   |
LL |     let _ = Command::new("sed");
   |                          ^^^^^
   |
   = help: use `regex`, or the methods on `str` instead

error: unnecessarily shelling out for trivial operation
  --> tests/ui/unnecessary_shell_command.rs:14:26
   |
LL |     let _ = Command::new("jq");
   |                          ^^^^
   |
   = help: use `serde_json`, or another native json parser instead

error: aborting due to 5 previous errors

